server/jetstream_benchmark_interest_stream_limit_test.go

```diff
@@ -0,0 +1,310 @@
+// Copyright 2022 The NATS Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build !skip_js_tests && !skip_js_cluster_tests && !skip_js_cluster_tests_2
+// +build !skip_js_tests,!skip_js_cluster_tests,!skip_js_cluster_tests_2
+
+package server
+
+import (
+	"fmt"
+	"math/rand"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/nats-io/nats.go"
+)
+
+func BenchmarkJetStreamInterestStreamWithLimit(b *testing.B) {
```

Comment:
Should we move these over to test directory, starting to get crowded in this directory with very long named tests etc.

Reply:
Do we have a sense of the private server functions they need access to?

Reply:
Sorry that was ambiguous. No private server symbols are used.
Only private test helpers such as `RunBasicJetStreamServer` or `createJetStreamClusterExplicit`.

I'll try to move these benchmarks, maybe it's easier than i expect.